File: 2356-largest-combination-with-bitwise-and-greater-than-zero/2356-largest-combination-with-bitwise-and-greater-than-zero-SET_BIT_ARRAY.py
class Solution:
    def largestCombination(self, candidates: List[int]) -> int:
        setBit=[0]*24
        for num in candidates:
            for i in range(24):
                if (num & 1<<i)!=0:
                    setBit[i]+=1
        return max(setBit)

File: 2356-largest-combination-with-bitwise-and-greater-than-zero/2356-largest-combination-with-bitwise-and-greater-than-zero-SET_BITARRAY.java
class Solution {
    public int largestCombination(int[] candidates) {
        ArrayList<Integer> setBit=new ArrayList<>(24);
        for (int i = 0; i < 24; i++) {
            setBit.add(0); 
        }
        System.out.println(setBit);
        for(int num:candidates){
            for(int i=0;i<24;i++){
                if((num&(1<<i))!=0){
                    setBit.set(i,setBit.get(i)+1);
                }
            }
        }
        return Collections.max(setBit);
    }
}
